### ISSUE TITLE
NSE table does not compile with CUDA

### ISSUE THREAD

**Issue by zingale** [AMReX-Astro/Microphysics#1391]
When compiling we get:

```
make: *** Waiting for unfinished jobs....
/ccs/home/zingale/Microphysics/nse_tabular/nse_table.H(246): error: identifier "nse_table_size::logrho_min" is undefined in device code

/ccs/home/zingale/Microphysics/nse_tabular/nse_table.H(246): error: identifier "nse_table_size::logrho_max" is undefined in device code

/ccs/home/zingale/Microphysics/nse_tabular/nse_table.H(248): error: identifier "nse_table_size::logT_min" is undefined in device code

/ccs/home/zingale/Microphysics/nse_tabular/nse_table.H(248): error: identifier "nse_table_size::logT_max" is undefined in device code

/ccs/home/zingale/Microphysics/nse_tabular/nse_table.H(250): error: identifier "nse_table_size::ye_min" is undefined in device code

/ccs/home/zingale/Microphysics/nse_tabular/nse_table.H(250): error: identifier "nse_table_size::ye_max" is undefined in device code
```

**Comment by zingale** [comment]
reproduce on groot in `Microphysics/unit_test/burn_cell_sdc/`

```
make USE_NSE_TABLE=TRUE USE_CUDA=TRUE -j 20
```

**Comment by zingale** [comment]
we compile fine with ROCm

**Comment by zingale** [comment]
this is fixed in #1395 
it seems to be a compiler bug
